File: backend/domain/model/comment.go
package model

import "time"

type Comment struct {
	ID        int
	PostID    int
	UserID    int
	UserName  string
	Body      string
	CreatedAt time.Time
	UpdatedAt time.Time
}

func (c Comment) IsEmpty() bool {
	return c.ID == 0
}

File: frontend/src/api/model/comment.ts
export interface Comment {
    id: number
    post_id: number
    user_id: number
    body: string
    created_at: string
    updated_at: string
    }

export interface CommentList {
    comments: Comment[]
}

export interface CreatComment{
    post_id: number
    user_id: number
    body: string
}

export interface UpdateComment{
    id: number
    post_id: number
    user_id: number
    body: string
}